### ISSUE TITLE
yaAGS depends on yaAGC

### ISSUE THREAD

**Issue by ThymoNL** [virtualagc/virtualagc#1133]
yaAGS currently includes agc_engine.h making it dependent on yaAGC. This presents problems for getting NASSP in line with virtualagc without having to modify source files. NASSP doesn't have the same file structure as virtualagc so when yaAGS looks to include "../yaAGC/agc_engine.h" it won't be able to find it.

I've been thinking about how to best go about removing this dependency but haven't reached a conclusion yet. One thing is certain and that is that shared code should be moved out of the agc_engine.h header and into its own file.

Question is, where should this file go?
Putting it inside the yaAGC and yaAGS folders isn't going to work because then one still depends on the other. That leaves two other options. Each component gets its own implementation, I'm not sure to what extend this might issues with interoperability between the two (maybe none?). The other is to create a place for things shared between yaAGC, yaAGS et al. Let's say yaLIB?

**Comment by rburkey2005** [comment]
Well, that's certainly a hassle.  I'm a little wary of making the kind of structural change suggested, because the (admittedly few!) people who know how to work with this code would now have to relearn.  It has been a long time since I've done anything substantive with yaAGS, so I'm not sure my opinions are truly meaningful, but back when I was working on it, here's what I would have done instead:

1. Remove the '#include "../yaAGC/agc_engine.h"' from aea_engine.h.
2. Throughout the yaAGS/ folder, replace all '#include "../yaAGC/agc_engine.h"' with '#include "aea_engine.h"'.
3. By repeated builds and trial-and-error, move stuff from agc_engine.h to aea_engine.h until I had warning-free builds again.

Is there some reason why a strategy like that wouldn't work?

**Comment by rburkey2005** [comment]
And yes, some items in agc_engine.h would have to be duplicated in aea_engine.h, rather than simply moved there, but I can't imagine it would be very much or would cause very much harm to do so.

**Comment by Thymo-** [comment]
Yeah, that would work. They do share some structs and stuff that are the same but nothing used to interface between the two. I'll get on it.
